Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/springcloud/SpringCloudAppNodePresenter.java

```diff
@@ -66,24 +65,25 @@ public void onDeleteApp(final String appId) throws IOException {
     }
 
     public static void awaitAndMonitoringStatus(String appId, DeploymentResourceStatus originalStatus) throws IOException, InterruptedException {
+        String clusterId = getParentSegment(appId);
         ExecutorService executor = Executors.newSingleThreadExecutor();
         Future future = executor.submit(() -> {
```

Comment:
Shall we use rxjava here? Using rxjava notify SpringCloudStateManager to update after status change?

Reply:
No, this is a watching thread for certain app, if we listen to SpringCloudStateManager, who will publish the update event?